### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by enoqueJonas**
# TDD project

## This PR was raised to:
- Write unit tests using Rspec.
- Understand an existing codebase in a way that allows for adding tests.
- Write Rspec unit tests with mocks.
- Understand the TDD concept.

## What does this PR do:
- Create a class called Solver.
  - Create a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- Create a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").
- Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
  - When N is divisible by 3, return "fizz".
  - When N is divisible by 5, return "buzz".
  - When N is divisible by 3 and 5, return "fizzbuzz".
  - Any other case, return N as a string (e.g. say N is 7 then return "7").

## Improvements to be made:
- None

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -245,44 +144,23 @@ Feel free to check the [issues page](../../issues/).
...
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
 <!-- LICENSE -->
 
 ## 📝 License <a name="license"></a>
 
-This project is [MIT](./LICENSE) licensed.
-
-_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
+This project is [MIT](./MIT) licensed.
```
Comment (Kamba56) [inline README.md]: - [x] Kindly fox the link to your mit license. The link currently displays a 404 error. The link should display your MIT license file.

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      fizzbuzz = solver.fizzbuzz(3)
+
+      expect(fizzbuzz).to eql 'fizz'
+    end
+
+    it 'Should return number to string' do
+      solver = Solver.new
+      fizzbuzz = solver.fizzbuzz(7)
+
+      expect(fizzbuzz).to eql '7'
```
Comment (Kamba56) [inline spec/solver_spec.rb]: - [x] Great job implementing the fizzbuzz tests, kindly add a test that handles the `buzz` return statement. This is to ensure you test every part of your method.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @enoqueJonas, @sebalp100, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great use of Github workflow!👏👏👏
- [x] All linters passed
- [x] Great job using the TDD process

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sebalp100** [comment]
Changes required implemented 👍

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @enoqueJonas  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] No linter errors :white_check_mark:
- [x] Good descriptive commit messages :white_check_mark:
- [x] Descriptive pull request title and summary :white_check_mark: 

## Optional suggestions
- [ Optional ] I would like to suggest that you add a sub-section on your readme file under `Getting Started` that informs someone looking at your project how to setup and  run the tests. Please see the example i've provided below for more details on this.

### Run tests
to run the tests, please follow the instructions below
1. Install dependencies
```sh
  $ bundle install
```
1. Run tests
```sh
$ rspec spec
``` 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**